x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorFilteringAction.java

```diff
@@ -70,8 +96,25 @@ public ActionRequestValidationException validate() {
                 validationException = addValidationError("[connector_id] cannot be [null] or [\"\"].", validationException);
             }
 
+            // If [filtering] is not present in the request payload it means that the user should define both [rules] and [advances_snippet]
```

Comment:
Why don't we allow to update both independent from each other? I could see users wanting to add one basic rule without changing advanced rules (they couldn't even be supported for their connector).

Reply:
Added logic to allow! :)